src/tic-tac-toe.cpp

```diff
@@ -29,6 +31,43 @@ class Player {
     bool d_isCross;
 };
 
+class Field {
+public:
+    Field (int n) {
+        for (int i = 1; i <= n; ++i) {
+            d_grid.push_back({});
+            for (int j = 1; j <= n; ++j) {
+                d_grid.back().push_back(nullptr);
+            }
+        }
+    }
+
+    bool makeMove(const Player& player, int x, int y) {
+        if (d_grid[x][y] == nullptr) {
+            d_grid[x][y] = &player;
+            return true;
+        }
+        return false;
+    }
+
+    Status status(const Player& player) { // 2 -- win, -2 -- fail, 0 -- the game in progress 1 -- draw, -1 -- other;
+        for (int i = 1; i <= d_grid.size(); ++i) {
+            for (int j = 1; j <= d_grid.size(); ++j) {
+                if (d_grid[i][j] == nullptr) {
```

Comment:
1. e_WIN the player who first gets n stones of their own color in a row, horizontally, vertically, or diagonally
2. `e_DRAW = !e_WIN && d_grid[x][y] != nullptr`
3. `e_IN_PROGRESS = !e_WIN || d_grid[x][y] == nullptr`

```
if (e_WIN){
  return e_WIN;
}
else if (d_grid[x][y] != nullptr){
   return e_DRAW;
}
else {
   return e_IN_PROGRESS;
}
```